src/parser/parser_support.c

```diff
@@ -9887,3 +9898,255 @@ pt_has_name_oid (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *contin
+
+  if (strlen (original_name) >= DB_MAX_IDENTIFIER_LENGTH - DB_MAX_SCHEMA_LENGTH)
+    {
+      char *name = (char *) original_name;
+      strcpy (name + 60, "...");
+      PT_ERRORf2 (parser, node,
+		  "Identifier name [%s] not allowed. It cannot exceed %d bytes.",
+		  original_name, DB_MAX_IDENTIFIER_LENGTH - DB_MAX_USER_LENGTH);
+    }
+
+  /* In case 1, 2, 3, 5 */
+  user_specified_name = pt_append_string (parser, NULL, downcase_resolved_name);
+  user_specified_name = pt_append_string (parser, user_specified_name, ".");
+  user_specified_name = pt_append_string (parser, user_specified_name, original_name);
+
+  if (PT_IS_NAME_NODE (node))
+    {
+      node->info.name.original = user_specified_name;
+      node->info.name.resolved = NULL;
+    }
+  else
+    {
+      assert (PT_IS_EXPR_NODE (node) && PT_IS_SERIAL (node->info.expr.op));
+
+      if (PT_IS_DOT_NODE (node->info.expr.arg1))
+	{
+	  node->info.expr.arg1->info.dot.arg2->info.name.original = user_specified_name;
+	  node->info.expr.arg1->info.dot.arg2->info.name.resolved = NULL;
+
+	  parser_free_tree (parser, node->info.expr.arg1->info.dot.arg1);
+	  node->info.expr.arg1 = node->info.expr.arg1->info.dot.arg2;
+	}
+      else
+	{
+	  assert (PT_IS_NAME_NODE (node->info.expr.arg1));
+
+	  node->info.expr.arg1->info.name.original = user_specified_name;
+	  node->info.expr.arg1->info.name.resolved = NULL;
+	}
+    }
+
+  if (current_user_name)
+    {
+      db_string_free (current_user_name);
+      current_user_name = NULL;
+    }
+
+  return node;
+}
+
+const char *
+pt_get_qualifier_name (PARSER_CONTEXT * parser, PT_NODE * name)
+{
+  char *dot = NULL;
+  const char *original_name = NULL;
+  const char *resolved_name = NULL;
+  const char *qualifier_name = NULL;
+
+  if (name == NULL || name->node_type != PT_NAME)
+    {
+      return NULL;
+    }
+
+  if (name->info.name.original == NULL || name->info.name.original[0] == '\0')
+    {
+      return NULL;
+    }
+
+  original_name = name->info.name.original;
+  resolved_name = name->info.name.resolved;
+
+  dot = (char *) strchr (original_name, '.');
+  if (dot)
+    {
+      dot[0] = '\0';
+      qualifier_name = pt_append_string (parser, NULL, original_name);
+      dot[0] = '.';
+
+      return qualifier_name;
+    }
+
+  return resolved_name;
+}
+
+const char *
+pt_get_name_without_current_user_name (const char *name)
+{
+  char *dot = NULL;
+  char *current_user_name = NULL;
+  const char *object_name = NULL;
+
+  if (name == NULL || name[0] == '\0')
+    {
+      return name;
+    }
+
+  dot = (char *) strchr (name, '.');
+  if (dot == NULL)
+    {
+      /*
+       * e.g.        name: object_name
+       *      object_name: object_name
+       */
+      return name;
+    }
+
+  current_user_name = db_get_user_name ();
+  assert (current_user_name != NULL);
+
+  dot[0] = '\0';
+
+  if (name && current_user_name && intl_identifier_casecmp (name, current_user_name) == 0)
```

Comment:
It does not come this far when name is null. The name check seems unnecessary.

Reply:
Okay. I'll remove the unnecessary null check for name, current_user_name.